firefox-ios/Client/Frontend/Browser/Toolbars/Redux/AddressBarState.swift

```diff
@@ -192,13 +192,14 @@ struct AddressBarState: StateType, Equatable {
 
         case ToolbarActionType.urlDidChange:
             guard let toolbarAction = action as? ToolbarAction else { return state }
+            let borderPosition = getAddressToolbarBorderPosition(action: toolbarAction, state: state)
```

Comment:
I like that we're using an existing action, but also curious on why not moving the logic in the middleware and can pass in the border position using
`let borderPosition = action.addressBorderPosition`

Reply:
This would require a `urlDidChange` middleware action, right? This was mainly to avoid adding an unnecessary middleware action for something that didn't technically need (i.e. no dependencies required) to go through the middleware, but I understand wanting to have all of the border logic in once place. Do you think it makes sense to add a middleware action here?

Reply:
We could also send two actions: the existing `urlDidChange` that goes to the reducer and one to the middleware that updates only the border. This way we don't duplicate code. 

The concerns I have about this code being in multiple places is that it renders the code we have in `DefaultToolbarManager` and it's unit tests useless and it's not the source of truth anymore. And we can't call the `DefaultToolbarManager` from the reducer as it's not supposed to have dependencies.

Reply:
I have added `ToolbarMiddlewareActionType.urlDidChange` so that the middleware continues to be the only one who handles border position logic therefore this logic is now removed from the `AddressBarState`'s reducer. The downfall is that we now have to dispatch both `ToolbarMiddlewareActionType.urlDidChange` and `ToolbarActionType.urlDidChange` in tandem 😪